todo_app/app.py

```diff
@@ -1,26 +1,77 @@
 from flask import Flask, redirect, render_template, request
-from todo_app.data.mongo_items import add_card, get_cards, get_lists, move_card, read_env_deatils
+from todo_app.data.mongo_items import add_card, get_cards, get_lists, move_card, read_env_deatils, get_myrole, get_currentuser, role_required
 from todo_app.flask_config import Config
 from todo_app.View_Class import ViewModel
+from flask_login import LoginManager, login_required, login_user
+from os import getenv
+import requests
+from todo_app.data.user_class import User
+from datetime import timedelta
 
 def create_app():
     read_env_deatils()
     app = Flask(__name__)
     app.config.from_object(Config())
+    login_manager = LoginManager()
+    CLIENTID = getenv('CLIENTID')
+    CLIENTSECRET = getenv('CLIENTSECRET')
+    BASEURL = getenv('URL')
+    STATE = getenv('STATE')
+    app.config['LOGIN_DISABLED'] = getenv('LOGIN_DISABLED') == 'True'
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        url = f"https://github.com/login/oauth/authorize?client_id={CLIENTID}&state={STATE}&redirect_uri={BASEURL}/login/callback&allow_signup=true"
+        return redirect(url)
+    
+    @login_manager.user_loader
+    def load_user(user_id):
+        user_name = user_id.split(",")[0].split(":")[1].replace("'","")
+        user_id = user_id.split(",")[1].split(":")[1].replace("'","")
+        thisuser = User(user_id,user_name)
+        return (thisuser)
+    login_manager.init_app(app)
 
     @app.route('/')
+    @login_required
     def index():
         item_view_model = ViewModel(get_cards())
-        return render_template('index.html', view_model=item_view_model, lists=get_lists())
+        ThisUser = get_currentuser()
+        return render_template('index.html', view_model=item_view_model, lists=get_lists(), user1=ThisUser, role=get_myrole(ThisUser.id))
 
     @app.route('/new', methods=['POST'])
+    @login_required
+    #@role_required('writer')
     def new():
+        if role_required('writer') :
```

Comment:
This isn't working - it lets me through regardless of my role being disabled! I think renaming this to be clearer might help, currently this returns True only if:
* The user role is blank & you've asked for writer
* But my user role is "disabled"

But maybe it would be better as `if !is_writer():` so that it's easy to reason about

Then the function could just be:
```
def is_writer():
    ThisUser = get_currentuser()
    UserRole = get_myrole(ThisUser.id)
    return UserRole == "writer"
```

Reply:
Implimented